lib/ansible/executor/module_common.py

```diff
@@ -634,39 +634,40 @@ def __repr__(self):
 
 
 class CollectionModuleInfo(ModuleInfo):
-    def __init__(self, name, paths):
+    _COLLECTION_PKG_RE = re.compile(to_text(r'ansible_collections\.([^\.]+)\.([^.]+)\.(.+)'))
```

Comment:
Could you please add more semantical meaning by naming the groups?
```suggestion
    _COLLECTION_PKG_RE = re.compile(to_text(r'ansible_collections\.(?P<coll_ns>[^\.]+)\.(?P<coll_name>[^.]+)\.(?P<py_mod_path>.+)'))
```